### PR TITLE
neighbor advertiser script

### PR DESCRIPTION
**Pull request by trzhang-msft**
neighbor advertiser script
- initiate API call to Ferret server to set up mappings
- set up mirror session
- set up vxlan session

### CODE REVIEW

File: scripts/neighbor_advertiser [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+MIRRORING_CONFIG_PATH = '/tmp/neighbor_advertiser/' + MIRRORING_CONFIG_NAME
+FERRET_NEIGHBOR_ADVERTISER_API_PREFIX = '/Ferret/NeighborAdvertiser/Slices/'
+VXLAN_TUNNEL_NAME = 'neighbor_advertiser_tunnel'
+MIRROR_SESSION_NAME = 'neighbor_advertiser_session'
+MIRROR_TABLE_NAME = 'acl_table_mirror_na'
+MIRROR_RULE_NAME = 'acl_rule_mirror_na'
+
+
+def get_switch_name():
+    hostname = subprocess.check_output(['hostname'])
```
Comment (pavel-shirshov) [inline scripts/neighbor_advertiser]: No need to run `hostname`. You can find this information in the file system.
```
def read_line(filename):
    ret = ''
    with open(filename) as f:
        ret = f.read()
    return ret.strip()

hostname = read_line('/etc/hostname')
```

File: scripts/neighbor_advertiser [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+    url = 'http://' + ferret_service_ip + ':85' + FERRET_NEIGHBOR_ADVERTISER_API_PREFIX + get_switch_name()
+    slice_obj = construct_neighbor_advertiser_slice()
+
+    response = requests.post(url, json = slice_obj)
+
+    ferret_ipv4_addr = ''
+
+    if response.ok:
+        na_configuration = json.loads(response.content)
+        ferret_ipv4_addr = na_configuration['ipv4Addr']
```
Comment (pavel-shirshov) [inline scripts/neighbor_advertiser]: if response is not ok?

File: scripts/neighbor_advertiser [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+            'src_ip': lo_inet_addr,
+            'dst_ip': ferret_ip
+        }
+    }
+
+    vxlan_tunnel_maps = {}
+    vlan_interfaces = get_vlan_interfaces()
+    index = 0
+
+    for vlan_interface in vlan_interfaces:
```
Comment (pavel-shirshov) [inline scripts/neighbor_advertiser]: you could use `enumerate` here to have an index of the element and traversed object element

File: scripts/neighbor_advertiser (hunk 1 above)
Comment (mslocrian) [reply scripts/neighbor_advertiser]: Or why not socket.gethostname()?

**Pull request closed without merging**